src/components/form.js

```diff
@@ -1,23 +1,71 @@
-import React from 'react';
+import React, { useRef } from 'react';
 import 'bootstrap/dist/css/bootstrap.min.css';
 import './books.css';
+import { useDispatch } from 'react-redux';
+import { nanoid } from '@reduxjs/toolkit';
+import { addBook } from '../redux/books/books';
 
-const Form = () => (
-  <>
-    <div className="border border-1 my-4" />
-    <section>
-      <h3>Add new book</h3>
-      <form className=" d-block d-md-flex gap-5 container-fluid m-0 p-0">
-        <input placeholder="Book title " className="Book-title" />
-        <input placeholder="Add Author " className="Book-title" />
-        <select className="dropdown">
-          <option value="action">action</option>
-          <option value="Science Fiction">Science Fiction</option>
-          <option value="Economy">Economy</option>
-        </select>
-        <button type="submit" className="primary-button-big border-0 rounded p-2 add-btn">ADD BOOK</button>
-      </form>
-    </section>
-  </>
-);
+const Form = () => {
+  // const [title, setTitle] = useState('');
+  // const [author, setAuthor] = useState('');
+
+  const dispatch = useDispatch();
+  const titleInput = useRef();
+  const authorInput = useRef();
+
+  const handleAddingBook = (e) => {
+    e.preventDefault();
+    const title = titleInput.current.value;
+    const author = authorInput.current.value;
+    if (title && author) {
+      const book = {
+        id: nanoid(),
+        title,
+        author,
+      };
+      dispatch(addBook(book));
+    }
+    titleInput.current.value = '';
+    authorInput.current.value = '';
+  };
+
+  // changing title
+  // const handleTitleChange = (e) => (setTitle(e.target.value));
+
+  // changing author
+  // const handleAuthorChange = (e) => {
+  //   setAuthor(e.target.value);
+  // };
+  return (
+    <>
+      <div className="border border-1 my-4" />
+      <section>
+        <h3>Add new book</h3>
+        <form
+          className=" d-block d-md-flex gap-5 container-fluid m-0 p-0 pb-4"
+          onSubmit={handleAddingBook}
+        >
+          <input
+            ref={titleInput}
+            placeholder="Book title "
+            className="Book-title"
+            // onChange={handleTitleChange}
+          />
+          <input
+            ref={authorInput}
+            placeholder="Add Author "
+            className="Book-title"
+            // onChange={handleAuthorChange}
```

Comment:
- [ ] [REQUIRED] To keep your code clean kindly remove any commented code from your repo 🙏 

__[Optional suggestion] :_ there's no need to remove the states and use `useRef();` , you can simply keep it as it was and the handle book can use that states like :_ 
```JS
const handleAddingBook = (e) => {
    e.preventDefault();
    if (title && author) {
      const book = {
        id: nanoid(),
        title:,
        author,
      };
      dispatch(addBook(book));
    }
    setTitle('')
    setAuthor('');
  };
```